frontend_tests/node_tests/stream_data.js

```diff
@@ -479,18 +479,20 @@ run_test('stream_settings', () => {
     assert.equal(sub_rows[2].invite_only, false);
 
     assert.equal(sub_rows[0].history_public_to_subscribers, true);
-    assert.equal(sub_rows[0].is_announcement_only, true);
+    assert.equal(sub_rows[0].stream_post_policy ===
+        stream_data.stream_post_policy_values.admins.code, true);
 
     const sub = stream_data.get_sub('a');
     stream_data.update_stream_privacy(sub, {
         invite_only: false,
         history_public_to_subscribers: false,
     });
-    stream_data.update_stream_announcement_only(sub, false);
+    stream_data.update_stream_post_policy(sub, 1);
     stream_data.update_calculated_fields(sub);
     assert.equal(sub.invite_only, false);
     assert.equal(sub.history_public_to_subscribers, false);
-    assert.equal(sub.is_announcement_only, false);
+    assert.equal(sub.stream_post_policy ===
+        stream_data.stream_post_policy_values.everyone.code, true);
 
```

Comment:
I think this should be replaced by `assert.equal(sub.stream_post_policy,stream_data.stream_post_policy_values.everyone.code)`.